tests/Sabre/Xml/ParserTest.php

```diff
@@ -0,0 +1,587 @@
+<?php
+
+namespace Sabre\Xml;
+
+/**
+ * Test the XML reader.
+ * -Test with valid document
+ * -Test XML schema validation
+ * -Test invalid documents (missing start and end tag, malformed start and end tag and malformed quotes)
+ *
+ *
+ * PHP version 5
+ *
+ * Class ParserTest
+ * @author     Daan Biesterbos <daanbiesterbos@gmail.com>
+ * @license http://sabre.io/license/ Modified BSD License
+ * @since     1.0.0
+ */
+class ParserTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * Simple test to check if the parser is able to parse documents under normal circumstances.
+     * Check if the result is as expected.
+     * - Use multiple schema's and multiple namespaces
+     * - Test attributes
+     * - Test elements with child nodes
+     * - Test element without child nodes and attributes ( <movie:bestseller /> )
+     */
+    function testParserShouldSucceed()
+    {
+        $reader = new Reader();
+        $reader->XML($this->xmlValidDocument);
+        $result = $reader->parse();
+        $this->assertNotEmpty($result);
+        $result = $result['value'];
+
+        // Validate store name
+        $this->assertEquals('Awesome Shop Inc.', $result[0]['value'], "The store name is not valid.");
```

Comment:
Could the entire `$result` array just be checked for as a whole. This will make this a _lot_ clearer I think.